commands/template_store_list.go

```diff
@@ -0,0 +1,175 @@
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// The URL where the store resides
+	DefaultTemplateList = "https://raw.githubusercontent.com/martindekov/store/martindekov/unofficial_templates_json/templates.json"
+	incubatorRepo       = "openfaas-incubator"
+	officialRepo        = "openfaas"
+	allTemplates        = "all"
+)
+
+var (
+	source string
+)
+
+func init() {
+	templateStoreListCmd.Flags().StringVarP(&source, "source", "s", allTemplates, "The source of the templates")
+	templateStoreListCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Shows additional language and platform")
+
+	templateStoreCmd.AddCommand(templateStoreListCmd)
+}
+
+var templateStoreListCmd = &cobra.Command{
+	Use:     `list`,
+	Short:   `List templates from OpenFaaS organizations`,
+	Aliases: []string{"ls"},
+	Long:    `List templates supported by openfaas or openfaas-incubator organizations or both`,
+	Example: `  faas-cli template store list --source=openfaas --verbose
+  faas-cli template store ls --source=openfaas-incubator -v
+  faas-cli template store ls --source=all`,
+	RunE: runTemplateStoreList,
+}
+
+func runTemplateStoreList(cmd *cobra.Command, args []string) error {
+	templatesInfo, templatesErr := getTemplateInfo(DefaultTemplateList)
+	if templatesErr != nil {
+		return fmt.Errorf("error while getting templates info: %s\n", templatesErr)
+	}
+
+	formattedOutput := formatCommandOutput(templatesInfo)
+
+	fmt.Fprintf(cmd.OutOrStdout(), "%s", formattedOutput)
+
+	return nil
+}
+
+func getTemplateInfo(repository string) ([]TemplateInfo, error) {
+	req, reqErr := http.NewRequest(http.MethodGet, repository, nil)
+	if reqErr != nil {
+		return nil, fmt.Errorf("error while trying to create request to take template info: %s\n", reqErr.Error())
+	}
+
+	client := http.DefaultClient
+	res, clientErr := client.Do(req)
+	if clientErr != nil {
+		return nil, fmt.Errorf("error while requesting template list: %s\n", clientErr.Error())
+	}
+
+	if res != nil {
+		defer res.Body.Close()
+		if res.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("unexpected status code wanted: %d got: %d\n", http.StatusOK, res.StatusCode)
+		}
+
+		body, bodyErr := ioutil.ReadAll(res.Body)
+		if bodyErr != nil {
+			return nil, fmt.Errorf("error while reading data from templates body: %s\n", bodyErr.Error())
+		}
+
+		templatesInfo := []TemplateInfo{}
+		unmarshallErr := json.Unmarshal(body, &templatesInfo)
+		if unmarshallErr != nil {
+			return nil, fmt.Errorf("error while unmarshalling into templates struct: %s\n", unmarshallErr.Error())
+		}
+		return templatesInfo, nil
+	}
+	return nil, fmt.Errorf("error empty response body from: %s\n", DefaultTemplateList)
+}
+
+func formatCommandOutput(templates []TemplateInfo) string {
+	if source != allTemplates && source != incubatorRepo && source != officialRepo {
+		fmt.Printf("\n`The --source` flag needs to have value %s, %s or %s\n", allTemplates, incubatorRepo, officialRepo)
```

Comment:
I am not sure we need a source flag at all?

Reply:
removed the flag ✔️